test/e2e_kubeadm/networking_test.go

```diff
@@ -0,0 +1,154 @@
+		cc := getClusterConfiguration(f.ClientSet)
+		// Extract the networking.podSubnet
+		// Please note that this test assumes that the user does not alter network configs
+		// using the extraArgs option.
+		if _, ok := cc["networking"]; ok {
+			netCC := cc["networking"].(map[interface{}]interface{})
+			if ps, ok := netCC["podSubnet"]; ok {
+				// If podSubnet is not specified, podSubnet cases will be skipped.
+				// Note that kubeadm does not currently apply defaults for PodSubnet, so we skip.
+				podSubnetInKubeadmConfig = true
+				cidrs := strings.Split(ps.(string), ",")
+				if len(cidrs) > 1 {
+					dualStack = true
+				}
+			}
+			if _, ok := netCC["serviceSubnet"]; ok {
+				// If serviceSubnet is not specified, serviceSubnet cases will be skipped.
+				serviceSubnetInKubeadmConfig = true
+			}
+		}
+	})
+
+	Context("single-stack", func() {
+		Context("podSubnet", func() {
+			It("should be properly configured if specified in kubeadm-config", func() {
+				if dualStack {
+					framework.Skipf("Skipping because cluster is dual-stack")
+				}
+				if !podSubnetInKubeadmConfig {
+					framework.Skipf("Skipping because podSubnet was not specified in kubeadm-config")
+				}
+				cc := getClusterConfiguration(f.ClientSet)
+				if _, ok := cc["networking"]; ok {
+					netCC := cc["networking"].(map[interface{}]interface{})
+					if ps, ok := netCC["podSubnet"]; ok {
+						// Check that the pod CIDR allocated to the node(s) is within the kubeadm-config podCIDR.
+						nodes, err := f.ClientSet.CoreV1().Nodes().List(metav1.ListOptions{})
+						framework.ExpectNoError(err)
+						for _, node := range nodes.Items {
+							if !subnetWithinSubnet(ps.(string), node.Spec.PodCIDR) {
+								e2elog.Failf("failed due to node(%v) IP %v not inside configured pod subnet: %s", node.Name, node.Spec.PodCIDR, ps)
+							}
+						}
+					}
+				}
+			})
+		})
+		Context("serviceSubnet", func() {
+			It("should be properly configured if specified in kubeadm-config", func() {
+				if dualStack {
+					framework.Skipf("Skipping because cluster is dual-stack")
+				}
+				if !serviceSubnetInKubeadmConfig {
+					framework.Skipf("Skipping because serviceSubnet was not specified in kubeadm-config")
+				}
+				cc := getClusterConfiguration(f.ClientSet)
+				if _, ok := cc["networking"]; ok {
+					netCC := cc["networking"].(map[interface{}]interface{})
+					if ss, ok := netCC["serviceSubnet"]; ok {
+						// Get the kubernetes service in the default namespace.
+						// Check that service CIDR allocated is within the serviceSubnet range.
+						svc, err := f.ClientSet.CoreV1().Services("default").Get("kubernetes", metav1.GetOptions{})
+						framework.ExpectNoError(err)
+						if !ipWithinSubnet(ss.(string), svc.Spec.ClusterIP) {
+							e2elog.Failf("failed due to service(%v) cluster-IP %v not inside configured service subnet: %s", svc.Name, svc.Spec.ClusterIP, ss)
+						}
+					}
+				}
+			})
+		})
+	})
+	Context("dual-stack", func() {
```

Comment:
This needs to be wrapped `[Feature:dual-stack]`

Reply:
Done!